### PR TITLE
JAMES-3474 Email/set should handle state property

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ import java.time.format.DateTimeFormatter
 import java.util.Date
 
 import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
-import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.RestAssured.{`given`, requestSpecification, _}
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:28]: wildcard spotted

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 2]
```diff
@@ -6235,6 +6236,249 @@ trait EmailSetMethodContract {
            |}""".stripMargin)
   }
 
+  @Test
+  def newStateShouldBeUpToDate(server: GuiceJamesServer): Unit = {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:6273]: Does this test passes on the distributed version?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 3]
```diff
@@ -6235,6 +6236,249 @@ trait EmailSetMethodContract {
...
+           |  "urn:ietf:params:jmap:mail"],
+           |  "methodCalls": [
+           |    ["Email/get", {
+           |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |      "ids":[]
+           |    }, "c1"]
+           |  ]
+           |}""".stripMargin)
+      .post
+      .`then`()
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:6475]: indent

Comment (Arsnael) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:6475]: Not resolved? :)

**Review by Arsnael — APPROVED** [review]

**Comment by LanKhuat** [comment]
test this please

**Review by remk — APPROVED** [review]
Do we miss a test with multiple changes, showing that the new state we get is the last one ?

**Comment by LanKhuat** [comment]
> Do we miss a test with multiple changes, showing that the new state we get is the last one ?

`
@Test
  def newStateShouldBeUpToDate(server: GuiceJamesServer): Unit = {
`
We have this test where we assert that the EmailChanges call returns nothing, which also means the newState is the latest one. I get that you want to explicitly assert the "newState" properties?

I don't really get the impact of multiple changes on this though... If there are less changes than the default limit (5 currently) then 5 changes should not bring any difference than 1 change. If there are more than 5 then of course newState won't be the latest one.

**Comment by Arsnael** [comment]
I will merge this. I propose you keep discussing it and if need to add a test, a new PR can be opened :)

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**